### PR TITLE
DEV: failed task output hash; user can verify that eng network output…

### PR DESCRIPTION
**Pull request by apalepu23**
1. failed task output hash
2. user can verify eng network output matches output hash stored on ethereum

### CODE REVIEW

**Review by lenak25 — CHANGES_REQUESTED** [review]
Thanks @apalepu23.
I've added some comments. 
In addition, please add the output hash to the `deploySecretContractFailureImpl` as well.

**Review by lenak25 — CHANGES_REQUESTED** [review]

**Review by lacabra — APPROVED** [review]
Everything looks good so far, other than the fact that CI breaks now because the P2P is out of sync with the number of parameters that are being passed to `CommitTaskFailure`, which is to be expected, and that @lenak25 will patch in due time:

```
p2p_1       | [Thu Sep 12 2019 02:48:36 GMT+0000 (Coordinated Universal Time)] INFO [COMMIT_RECEIPT] success for commitTaskFailure of task eebc2fa94431bd83d354ad3cab16435dc8e25fec9b3440a74564d259e347f28c receipt = [object Promise]
p2p_1       | (node:301) UnhandledPromiseRejectionWarning: Error: Invalid number of parameters for "commitTaskFailure". Got 4 expected 5!
```

File: enigma-js/src/Enigma.js [hunk 1]
```diff
@@ -496,6 +496,44 @@ export default class Enigma {
...
+   * Verify ENG network status matches status registered on ETH
+   *
+   * @param {Task} task - Task wrapper for contract deployment and compute tasks
+   * @return {boolean} True/false on whether statuses match
+   */
+  async verifyTaskStatus(task) {
+    const ethStatus = (await this.getTaskRecordStatus(task)).ethStatus;
+    switch (task.engStatus) {
+      case 'SUCCESS':
+        return ethStatus === 2;
```
Comment (lenak25) [inline enigma-js/src/Enigma.js]: Could you please use constants instead?

**Review by lenak25 — CHANGES_REQUESTED** [review]

File: enigma-js/src/Enigma.js (hunk 1 above)
Comment (lacabra) [reply enigma-js/src/Enigma.js]: Implemented in dcde0fb, so much better, thanks for the suggestion.

**Review by lenak25 — APPROVED** [review]
LGTM

**Pull request merged**